src/test/e2e/38_ephemeral_container_test.go

```diff
@@ -0,0 +1,73 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2021-Present The Zarf Authors
+
+// Package test provides e2e tests for Zarf.
+package test
+
+import (
+	"fmt"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestEphemeralContainers(t *testing.T) {
+	t.Log("E2E: Ephemeral Containers mutation")
+
+	tmpdir := t.TempDir()
+
+	// we need to create a test package that contains the images we want to potentially use
+	// this should ideally be a single pod such that naming is static
+	stdOut, stdErr, err := e2e.Zarf(t, "package", "create", "src/test/packages/38-ephemeral-container", "-o", tmpdir, "--skip-sbom")
+	require.NoError(t, err, stdOut, stdErr)
+	packageName := fmt.Sprintf("zarf-package-basic-pod-%s-0.0.1.tar.zst", e2e.Arch)
+	path := filepath.Join(tmpdir, packageName)
+
+	// deploy the above package
+	stdOut, stdErr, err = e2e.Zarf(t, "package", "deploy", path, "--confirm")
+	require.NoError(t, err, stdOut, stdErr)
+
+	// using a pod the package deploys - run a kubectl debug command
+	stdOut, stdErr, err = e2e.Kubectl(t, "debug", "test-pod", "-n", "test", "--image=busybox:1.36", "--profile", "general")
+	require.NoError(t, err, stdOut, stdErr)
+
+	// there is no native 'wait' logic for ephemeral containers
```

Comment:
This is a bit "cludgy" - if there is a better pattern here - I am open to it. Otherwise there is some race condition between the kubectl `debug` command and `get` command such that it returns an empty string (possibly getting information too quickly before the mutation and admission has occurred).

Reply:
Looks like should work 
```bash
kubectl wait --for=condition=Ready pod/my-debug-pod --timeout=10s
```

Reply:
Definitely favor an await-style API over polling and checking 👍

Reply:
The issue here being that the pod status never changes and the use of `debug` is adding a `ephemeral container`  to an existing pod.